src/base-client-helpers_test.ts

```diff
@@ -0,0 +1,141 @@
+import { assertEquals } from "https://deno.land/std@0.185.0/testing/asserts.ts";
+import { _internals, getUserAgent } from "./base-client-helpers.ts";
+import { serializeData } from "./base-client-helpers.ts";
+import { assertSpyCalls, stub } from "./dev_deps.ts";
+
+Deno.test(_internals.getModuleVersion.name, async (t) => {
+  await t.step(
+    "should return the version if the module is sourced from deno.land",
+    () => {
+      const getModuleUrlStub = stub(_internals, "getModuleUrl", () => {
+        return new URL("https://deno.land/x/deno_slack_api@2.1.0/mod.ts)");
+      });
+
+      try {
+        const moduleVersion = _internals.getModuleVersion();
+
+        assertSpyCalls(getModuleUrlStub, 1);
+        assertEquals(moduleVersion, "2.1.0");
+      } finally {
+        getModuleUrlStub.restore();
+      }
+    },
+  );
+
+  await t.step(
+    "should return the unknown if the module is not sourced from deno.land",
+    () => {
+      const getModuleUrlStub = stub(_internals, "getModuleUrl", () => {
+        return new URL("file:///hello/world.ts)");
+      });
+      try {
+        const moduleVersion = _internals.getModuleVersion();
+
+        assertSpyCalls(getModuleUrlStub, 1);
+        assertEquals(moduleVersion, undefined);
+      } finally {
+        getModuleUrlStub.restore();
+      }
+    },
+  );
+
+  await t.step(
+    "should return the unknown if the module version is invalid",
```

Comment:
What's the idea in this test? How is the version invalid in the test case?

Reply:
Good point, I will update the test description, this test validates that the Regex used to extract the version works properly and would return `undefined` if the module url path does not match `deno_slack_api@`

Example `deno_slack_sdk`